perf_test/sparse/KokkosSparse_sptrsv.cpp

```diff
@@ -198,30 +254,50 @@ int test_sptrsv_perf(std::vector<int> tests, std::string& lfilename, std::string
     KernelHandle kh;
     bool is_lower_tri = true;
 
-    std::cout << "Create handle" << std::endl;
+    std::cout << "Create handle (lower)" << std::endl;
     switch(test) {
       case LVLSCHED_RP:
         kh.create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_RP, nrows, is_lower_tri);
         kh.get_sptrsv_handle()->print_algorithm();
         break;
       case LVLSCHED_TP1:
         kh.create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_TP1, nrows, is_lower_tri);
+        std::cout << "TP1 set team_size = " << team_size << std::endl;
+        if (team_size != -1) kh.get_sptrsv_handle()->set_team_size(team_size);
+        kh.get_sptrsv_handle()->print_algorithm();
+        break;
+      case LVLSCHED_TP1CHAIN:
+        printf("TP1 with CHAIN\n");
+        printf("chain_threshold %d\n", chain_threshold);
+        printf("team_size %d\n", team_size);
+        kh.create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHD_TP1CHAIN, nrows, is_lower_tri);
+        kh.get_sptrsv_handle()->reset_chain_threshold(chain_threshold);
+        if (team_size != -1) kh.get_sptrsv_handle()->set_team_size(team_size);
+        if (vector_length != -1) kh.get_sptrsv_handle()->set_vector_size(vector_length);
         kh.get_sptrsv_handle()->print_algorithm();
         break;
-/*
       case LVLSCHED_TP2:
         kh.create_sptrsv_handle(SPTRSVAlgorithm::SEQLVLSCHED_TP2, nrows, is_lower_tri);
+        if (team_size != -1) kh.get_sptrsv_handle()->set_team_size(team_size);
+        if (vector_length != -1) kh.get_sptrsv_handle()->set_vector_size(vector_length);
+        kh.get_sptrsv_handle()->print_algorithm();
+        break;
+      case CUSPARSE_K:
+        printf("CUSPARSE WRAPPER\n");
+        kh.create_sptrsv_handle(SPTRSVAlgorithm::SPTRSV_CUSPARSE, nrows, is_lower_tri);
         kh.get_sptrsv_handle()->print_algorithm();
         break;
-*/
-#ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
       case CUSPARSE:
```

Comment:
@ndellingwood could you please remind me what different between ```CUSPARSE``` and ```CUSPARSE_K```?

Reply:
Good question - CUSPARSE_K is the kokkos-kernels supported version I added in the PR, CUSPARSE is the hand-rolled implementation within the perf test.

Reply:
How do CUSPARSE_K and CUSPARSE compare in terms of perf? Should we still need to keep CUSPARSE case in the perf test when we already have CUSPARSE_K?

Reply:
I haven't compared carefully, they should be nearly the same. It may be interesting to keep for comparison, but I can also remove to clean up the perf test.

Reply:
Understand. It is up to you to remove it or not.

Reply:
Thanks @vqd8a for the catch, I'll leave it in for now to comparison test but will clean up in a PR shortly after.